### DISCUSSION TITLE
Portable ruby issue on previously working brew install

### DISCUSSION THREAD

**Discussion by firmbeliever** [Homebrew/discussions#1363] [Everyday usage] [answered]
Hi, 
I am currently running mac 10.15.7 - Catalina.

I installed brew about a year and half ago, installed python, haven't used it much since.
Today I tried installing libomp. When I ran 'brew install libomp' I was getting sha mismatch errors related to:
portable-ruby-2.6.3_2.yosemite.bottle.tar.gz. 

The error message instructed me to remove the bottle.tar.gz file and rerun. I tried that multiple times, but each time it would just download the same file with the mismatched sha. Eventually I downloaded the file here (https://github.com/Homebrew/homebrew-portable-ruby/releases), checked its sha256, which matched what brew was expecting, as per the previous error messages.

So I placed that new bottle.tar.gz in /Users/me/Library/Caches/Homebrew. I'm not getting the sha mismatch errors anymore, so I think that issue is fixed, but now basically any brew command I run results in this error:

bash-3.2$ brew cleanup
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:b065e5e3783954f3e65d8d3a6377ca51649bfcfa21b356b0dd70490f74c6bd86
Already downloaded: /Users/wl1037923/Library/Caches/Homebrew/portable-ruby-2.6.3_2.yosemite.bottle.tar.gz
==> Pouring portable-ruby-2.6.3_2.yosemite.bottle.tar.gz
Error: Failed to install ruby 2.6.3_2!
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
bash-3.2$ 

That's the whole thing - no indication as to why it failed. I get this error if I run brew cleanup, brew doctor, brew update, brew upgrade, etc.

Any ideas on what the issue is?
Thanks.

**Comment by carlocab** [comment, marked as answer]
Sounds a lot like this issue: https://github.com/Homebrew/homebrew-core/issues/75997

Try examining the file with the mismatched checksum with `file`. Chances are it's not actually a tarball, and that your download was intercepted, possibly by some firewall.

**Reply by firmbeliever** [reply]
Just to be clear, I'm no longer getting the mismatched checksum error, so I don't have that file anymore.

But when I got the sha256 of my downloaded tar.gz, it exactly matched what brew had said it was expecting in the earlier error messages. That wouldn't be the case if the file was intercepted, right?

Here's my results from running `file`:

bash-3.2$ file /Users/wl1037923/Library/Caches/Homebrew/portable-ruby-2.6.3_2.yosemite.bottle.tar.gz 
/Users/wl1037923/Library/Caches/Homebrew/portable-ruby-2.6.3_2.yosemite.bottle.tar.gz: gzip compressed data, was "portable-ruby-bottle.tar", last modified: Wed Apr 17 05:24:24 2019, from Unix, original size modulo 2^32 28075520

**Reply by carlocab** [reply]
Hard to say what happened if you can no longer reproduce the problem then. We can revisit it if it comes up again.

**Reply by firmbeliever** [reply]
To clarify:
The tarball that brew was downloading had the mismatched sha.

So I went and manually downloaded the file from the URL I mentioned in my original post. Its sha exactly matches what brew was expecting, so that's what I'm using now and getting this very vague error. And that's what the `file` output above is based on.

**Reply by firmbeliever** [reply]
I'm no longer getting the mismatch error, I'm getting a different error, which is what I'm looking for help with

**Reply by carlocab** [reply]
I see. Not sure downloading the file manually and putting it in `HOMEBREW_CACHE` will work. (Ideally it should, but I'm not convinced that it necessarily will.) Try downloading it again, but using `brew`, and not manually.

**Reply by firmbeliever** [reply]
Thanks. I'm not a brew expert - can you tell me exactly what command I should run?

**Reply by Bo98** [reply]
`brew` does very little in the pour step.

If you are getting that error then it means this is failing:

```sh
tar xzf ~/Library/Caches/Homebrew/portable-ruby-2.6.3_2.yosemite.bottle.tar.gz
cd portable-ruby
./2.6.3_2/bin/ruby --version # fails
```

**Reply by firmbeliever** [reply]
@Bo98 Any thoughts on what causes that and how I can fix it?

**Reply by Bo98** [reply]
No idea. Security software? Maybe try the above commands in a temporary directory and see if it you get any errors?

**Reply by firmbeliever** [reply]
Ah, yes. I just downloaded another tarball and ran the above commands from my downloads folder. I get a pop-up saying that the developer can't be verified. So that must be what's blocking it. Will see how/if I can work around this. Thanks!

**Reply by carlocab** [reply]
Sounds like third-party security software. If you figure out it, please leave a note here about which one you use and how you fixed it in case others face the same problem.

**Reply by Bo98** [reply]
Ah yes if you downloaded through a browser you will need to run `xattr -c` on the file. This isn't a problem if downloaded through `brew`.

**Reply by carlocab** [reply]
...now I know why none of the bottles I download from GitHub Actions work without dealing with copious error messages from SIP. TIL.

**Reply by firmbeliever** [reply]
@Bo98 well, when brew downloaded it, that's when I was getting the mismatch sha errors, so I wonder if my company's security software was disrupting that download as well, as carlocab suggested.

**Reply by Bo98** [reply]
You can check by allowing brew to download it and seeing what the contents of `~/Library/Caches/Homebrew/portable-ruby-2.6.3_2.yosemite.bottle.tar.gz` is.

**Reply by firmbeliever** [reply]
Yes, my company's software blocked brew from accessing ghcr.io, so the "tarball" was really just some html from the page that it redirected to.

In my case, I can get around this by accessing the page in a browser and selecting that I want to proceed to the page anyway, which unblocks it and allows brew to then do its download.

I've now got some local permissions issue, but I don't think that's brew related, so will consider this resolved! Thanks to you both!
